pipeline/frontend/yaml/constraint/constraint.go

```diff
@@ -51,20 +50,26 @@ type (
 // Match returns true if all constraints match the given input. If a single
 // constraint fails a false value is returned.
 func (c *Constraints) Match(metadata frontend.Metadata) bool {
-	return c.Platform.Match(metadata.Sys.Arch) &&
+	match := c.Platform.Match(metadata.Sys.Arch) &&
 		c.Environment.Match(metadata.Curr.Target) &&
 		c.Event.Match(metadata.Curr.Event) &&
 		c.Branch.Match(metadata.Curr.Commit.Branch) &&
 		c.Repo.Match(metadata.Repo.Name) &&
 		c.Ref.Match(metadata.Curr.Commit.Ref) &&
 		c.Instance.Match(metadata.Sys.Host) &&
-		c.Matrix.Match(metadata.Job.Matrix) &&
-		c.Path.Match(metadata.Curr.Commit.ChangedFiles, metadata.Curr.Commit.Message)
+		c.Matrix.Match(metadata.Job.Matrix)
+
+	// changed files filter do not apply for tag event
+	if metadata.Curr.Event != frontend.EventTag {
+		match = match && c.Path.Match(metadata.Curr.Commit.ChangedFiles, metadata.Curr.Commit.Message)
+	}
```

Comment:
Not really sure if this is needed as part from the Github tag (push) event no tag event should even set the changedfiles property.

Reply:
but at the moment we do move that constraint logic into remote - by ensure there no ChangedFiles are set.

but constraint logic should be in one place

Reply:
tldr: yes we dont need it but it hardens woodpecker & make code more understandable

Reply:
Okay, I see that point. Should we change it to only be active on push and pull-request to be even more specific and safe for future events.

```suggestion
	// changed files filter do not apply for tag event
	if metadata.Curr.Event == frontend.EventPull || metadata.Curr.Event == frontend.EventPush {
		match = match && c.Path.Match(metadata.Curr.Commit.ChangedFiles, metadata.Curr.Commit.Message)
	}
```

Reply:
ACK